### PR TITLE
Split `go-ipfs/routing/record` into `go-libp2p-record`

### PR DESCRIPTION
**Pull request by geoah**
This PR splits `go-ipfs/routing/record` from `go-ipfs` into `libp2p/go-libp2p-record`

* Undoes gx rewrite
* Adds travis-ci
* Re-Adds `README` and `LICENCE` (messed up because of my graft)
* Removes `go-ipfs/path` dependency.

Note: This cannot be tested or published until `go-key` has been first published to gx.

@whyrusleeping if can please enable travis for this repo.
@whyrusleeping also if you could please disable the `Allow merge commits` for this repo or manually rebase this so the history remains intact.

Thanks :)

### CODE REVIEW

**Comment by whyrusleeping** [comment]
Hrm... this is tricky. I don't think we should be depending on the go-libp2p-kad-dht package here, especially since the go-libp2p-kad-dht package will need to import this package...

Maybe it makes more sense to move the protobuf stuff into this package? Or maybe the `Message` protobuf and the `Record` protobuf need to be split out into separate files?

**Comment by geoah** [comment]
@whyrusleeping `record` is now here, my the last commit is a bit weird so please check if you have a specific way you log before merging.

File: selection.go [hunk 1]
```diff
@@ -23,13 +21,13 @@ func (s Selector) BestRecord(k key.Key, recs [][]byte) (int, error) {
 
 	parts := strings.Split((string(k)), "/")
 	if len(parts) < 3 {
-		log.Infof("Record key does not have selectorfunc: %s", k)
+		logging.Logger(packageName).Infof("Record key does not have selectorfunc: %s", k)
```
Comment (whyrusleeping) [inline selection.go]: you can just do `log.Infof`. You've declared the logger as a global in validation.go

**Comment by whyrusleeping** [comment]
@geoah commented on the logging thing, otherwise this LGTM

**Comment by geoah** [comment]
@whyrusleeping my `goimports` was simply [misbehaving](https://github.com/golang/go/issues/7463) and really confused me.

Fixed now, please go ahead and merge this. :) thanks.

**Comment by whyrusleeping** [comment]
@geoah wonderful work! This is gonna be awesome.

**Pull request merged**